airbyte-integrations/connectors/source-airtable/source_airtable/auth.py

```diff
@@ -0,0 +1,56 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+from typing import Any, Mapping, Union
+
+import requests
+from airbyte_cdk.sources.streams.http.requests_native_auth import (
+    BasicHttpAuthenticator,
+    SingleUseRefreshTokenOauth2Authenticator,
+    TokenAuthenticator,
+)
+
+
+class AirtableOAuth(SingleUseRefreshTokenOauth2Authenticator):
+    """
+    https://airtable.com/developers/web/api/oauth-reference#token-expiry-refresh-tokens
+    """
+
+    def build_refresh_request_headers(self) -> Mapping[str, Any]:
+        """
+        https://airtable.com/developers/web/api/oauth-reference#token-refresh-request-headers
+        """
+        config_creds = self._connector_config["credentials"]
+        return {
+            "Authorization": BasicHttpAuthenticator(config_creds["client_id"], config_creds["client_secret"]).token,
+            "Content-Type": "application/x-www-form-urlencoded",
+        }
+
+    def build_refresh_request_body(self) -> Mapping[str, Any]:
+        """
+        https://airtable.com/developers/web/api/oauth-reference#token-refresh-request-body
+        """
+        return {
+            "grant_type": self.get_grant_type(),
+            "refresh_token": self.get_refresh_token(),
+        }
+
+    def _get_refresh_access_token_response(self) -> Mapping[str, Any]:
+        response = requests.request(
+            method="POST",
+            url=self.get_token_refresh_endpoint(),
+            data=self.build_refresh_request_body(),
+            headers=self.build_refresh_request_headers(),
+        )
+        response.raise_for_status()
+        return response.json()
+
+
+class AirtableAuth:
+    def __new__(self, config: dict) -> Union[TokenAuthenticator, AirtableOAuth]:
```

Comment:
Hi, 
Please pay your attention to the next points:
1. Any `__new__` magic function accepts an instance as the first argument. And it will be better if it is "cls".
2. `config["api_key"]`  can be empty or `config` can be `None`. I would have check by this if had written this condition: `(config or {}).get("api_key")`.